src/test/FuzzerImpl.cpp

```diff
@@ -574,6 +574,100 @@ attemptToApplyOps(LedgerTxn& ltx, PublicKey const& sourceAccount,
     }
 }
 
+// Unlike Asset, this can be a constexpr.
+struct AssetID
+{
+    constexpr AssetID() : mIsNative(true), mId(0)
+    {
+    }
+
+    constexpr AssetID(int id) : mIsNative(false), mId(id)
+    {
+    }
+
+    Asset
+    toAsset() const
+    {
+        return mIsNative ? txtest::makeNativeAsset()
+                         : FuzzUtils::makeAsset(mId);
+    }
+
+    bool const mIsNative;
+    int const mId; // meaningful only if !isNative
+};
+
+struct OfferParameters
+{
+    constexpr OfferParameters(AssetID const bid, AssetID const sell,
+                              int publicKey, int32_t priceNumerator,
+                              int32_t priceDenominator, int64_t amount)
+        : mBid(bid)
+        , mSell(sell)
+        , mPublicKey(publicKey)
+        , mNumerator(priceNumerator)
+        , mDenominator(priceDenominator)
+        , mAmount(amount)
+    {
+    }
+
+    AssetID const mBid;
+    AssetID const mSell;
+    int const mPublicKey;
+    int32_t const mNumerator;
+    int32_t const mDenominator;
+    int64_t const mAmount;
+};
+
+// In order for certain operation combinations to be valid, we need an
+// initial order book with various offers. The order book will consist
+// of identical setups for the asset pairs:
+//      XLM - A
+//      A   - B
+//      B   - C
+//      C   - D
+// For any asset A and asset B, the generic order book setup will be as
+// follows:
+//
+// +------------+-----+------+--------+------------------------------+
+// |  Account   | Bid | Sell | Amount | Price (in terms of Sell/Bid) |
+// +------------+-----+------+--------+------------------------------+
+// | 0          | A   | B    |     10 | 3/2                          |
+// | 1 (issuer) | A   | B    |     50 | 3/2                          |
+// | 2          | A   | B    |    100 | 1/1                          |
+// | 3 (issuer) | B   | A    |     10 | 10/9                         |
+// | 4          | B   | A    |     50 | 10/9                         |
+// | 0          | B   | A    |    100 | 22/7                         |
+// +------------+-----+------+--------+------------------------------+
+//
+// This gives us a good mixture of buy and sell, issuers with offers,
+// and an account with a bid and a sell. It also gives us an initial
+// order book that is in a legal state.
+std::array<OfferParameters, 24> constexpr orderBookParameters{
+    {OfferParameters{AssetID(), AssetID(1), 13, 3, 2, 10}, // asset pair 0
```

Comment:
I think we should try to  make this section look as much as possible like the comment above (and that comment should be removed as it's wrong anyways):
* use an empty line between order book pairs
* price should be the last argument, and price should be sorted so that we can clearly see how the order book is structured (note that because price is in terms of sell/bid, the sort order flips when switching buy and sell)
* I think you can get rid of the `OfferParameters`? (or maybe this is C++ 17)

Reply:
Each section we should have some comment that explains which property we're trying to achieve.

To do this, we may have to do the same for accounts, that right now are setup pretty much all the same (instead, we probably want to mix things up a bit in terms of limits on balance entries and assets trusted).

Note: as we're trying to match the coverage from master, we may need less accounts and offers variations than what we're doing now.

Reply:
> and that comment should be removed as it's wrong anyways

I left the part that documents the initial state of the order book, but removed the stuff about it being a "good mixture".

Reply:
> use an empty line between order book pairs

Done.

Reply:
> price should be the last argument, and price should be sorted so that we can clearly see how the order book is structured (note that because price is in terms of sell/bid, the sort order flips when switching buy and sell)

I've made Price into the last argument (and otherwise brought the argument order into line with the comment).  I think it was already in best-offer order?  I might be misunderstanding that.

Reply:
> I think you can get rid of the OfferParameters? (or maybe this is C++ 17)

Oh yes, you're right.  I thought it was C++17, but `clang` at least is allowing me to do it with `std=c++14`.

Reply:
> Each section we should have some comment that explains which property we're trying to achieve.
>
> To do this, we may have to do the same for accounts, that right now are setup pretty much all the same (instead, we probably want to mix things up a bit in terms of limits on balance entries and assets trusted).
>
> Note: as we're trying to match the coverage from master, we may need less accounts and offers variations than what we're doing now.

Agreed on all points -- I wanted to start with just the order book so that we could get consensus on how the parameter-refactoring should look, and then once that was merged, I'd do the same for accounts, and for the other structures that I'm adding to setup.  I also definitely plan to change the setup parameters themselves, but I wanted to separate the mechanism changes from the policy changes.

Reply:
I think this is good now. I'm going to merge it tomorrow

Reply:
> I think this is good now. I'm going to merge it tomorrow

Thanks.  I've rebased onto master.